### ISSUE TITLE
protoc-gen-gogofaster failed with status code 1 yet compiles successfully

### ISSUE THREAD

**Issue by orkunkl** [gogo/protobuf#601]
PROTOC VERSION: 3.7.1

We are using prototool and protobuf for our projects at https://github.com/iov-one/. In one of our [project](https://github.com/iov-one/weave-starter-kit) when we compile `.proto` files with protoc-gen-gogofaster this error is thrown:
```
<input>:1:1:994								intStringLen := int(stringLen)
<input>:1:1:995								if intStringLen < 0 {
<input>:1:1:996									return ErrInvalidLengthCodec
<input>:1:1:997								}
<input>:1:1:998								postIndex := iNdEx + intStringLen
<input>:1:1:999								if postIndex < 0 {
<input>:1:1:protoc-gen-gogofaster failed with status code 1.
make: *** [protoc] Error 255
```
Here is the full error: https://gist.github.com/orkunkl/15b2f9f8012991c05291cb576164f0e1

Error output seems like generated [codec.pb.go](https://github.com/iov-one/weave-starter-kit/blob/master/x/custom/codec.pb.go) of [codec.proto](https://github.com/iov-one/weave-starter-kit/blob/master/x/custom/codec.proto). What could be the reason?

**Comment by apelisse** [comment]
The output that you gave is sorted alphabetically rather than by line number which makes it very hard to follow :-)

Anyways, here's what I found.

This is the actual error:
```
<input>:1:1:2019/08/13 12:37:49 protoc-gen-gogo: error:bad Go source code was generated: 112:11: expected type, found '.' (and 10 more errors)
```

So let's find that line 112:
```
<input>:1:1:112	Metadata	*.Metadata	`protobuf:"bytes,1,opt,name=metadata,proto3" json:"metadata,omitempty"`
```

Yep,  that doesn't look good. @orkunkl Do you happen know what could cause that? I'll try to look at the code and see if I find something related.

**Comment by orkunkl** [comment]
We are using (uber/prototool)[https://github.com/uber/prototool] at IOV. I suspect [includes](https://orkunkl.github.com/iov-one/weave-starter-kit/blob/create-app/prototool.yaml#L8-L12) in *prototool.yaml* causing the bug.

```yaml
includes:
  - .
  - /usr/include
  - spec
  - spec/github.com/iov-one/weave
```
We use `gogoproto` to handle imports. `spec` folder is where we define weave framework(parent framework) protobuf definitions. Maybe including `spec` and `spec/github.com/iov-one/weave` at the same time causing error?

**Comment by apelisse** [comment]
Yeah, I don't know about that. It's certainly not related to the things I've done, @jmarais any idea?

**Comment by orkunkl** [comment]
@apelisse Solved the issue. You can check https://github.com/iov-one/weave-starter-kit/pull/37
